File: src/main/java/ru/serdtsev/homemoney/dto/Reserve.java
package ru.serdtsev.homemoney.dto;


import ru.serdtsev.homemoney.account.BalanceDto;

import java.math.BigDecimal;

public class Reserve extends BalanceDto {
  private BigDecimal target;

  public Reserve() {
  }

  public BigDecimal getTarget() {
    return target != null ? target : BigDecimal.ZERO;
  }

  public void setTarget(BigDecimal target) {
    this.target = target;
  }
}

File: src/main/java/ru/serdtsev/homemoney/dto/Reserve.kt
package ru.serdtsev.homemoney.dto

import java.math.BigDecimal

class Reserve : Balance {
  var target: BigDecimal?
    get() = field ?: BigDecimal.ZERO
  constructor(): this(null)
  constructor(target: BigDecimal?) {
    this.target = target
  }
}
